plugins/notifiers/client.go

```diff
@@ -7,19 +7,52 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/goto/guardian/pkg/evaluator"
 	"github.com/goto/guardian/pkg/log"
+	"github.com/goto/guardian/plugins/notifiers/lark"
+	"github.com/goto/guardian/plugins/notifiers/slack"
 	"github.com/mitchellh/mapstructure"
 
 	"github.com/goto/guardian/domain"
-	"github.com/goto/guardian/plugins/notifiers/slack"
 )
 
 type Client interface {
 	Notify(context.Context, []domain.Notification) []error
 }
 
+type NotifyManager struct {
+	clients []Client
+	configs []Config
+}
+
+func (m *NotifyManager) Notify(ctx context.Context, notification []domain.Notification) []error {
+	var errs []error
+	for i, client := range m.clients {
+		// evaludate criteria
+		config := m.configs[i]
+		v, err := evaluator.Expression(config.Criteria).EvaluateWithVars(map[string]interface{}{
+			"email": notification[0].User,
+		})
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		// if the expression evaluates to true, notify the client
+		if match, ok := v.(bool); !ok {
+			err = fmt.Errorf("notifier expression did not evaluate to a boolean: %s", config.Criteria)
+			errs = append(errs, err)
+		} else if match {
+			client.Notify(ctx, notification)
```

Comment:
need to accumulate the errors from `client.Notify`
```suggestion
			if notifyErrs := client.Notify(ctx, notification); notifyErrs != nil {
        errs = append(errs, notifyErrs...)
      }
```